src/app/services/swagger.service.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-console */
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { environment } from 'src/environments/environment';
+import { SwaggerSummary } from '../models/swagger-summary/swagger-summary';
+import { SwaggerUploadResponse } from '../models/swagger-upload-response/swagger-upload-response';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SwaggerService {
+  public uploadForm: FormGroup;
+
+  public secondsUntilETA: number;
+
+  public swaggerSummary: SwaggerSummary;
+
+  public swaggerUploadResponse: SwaggerUploadResponse;
+
+  private uri = `${environment.API_BASE_URL}:${environment.PORT}`;
+
+  constructor(private http: HttpClient) { }
+
+  async uploadSwaggerFile(formData: FormData): Promise<SwaggerUploadResponse> {
+    const swaggerResponse = await this.http.post<SwaggerUploadResponse>(`${this.uri}/Docutest/upload`, formData).toPromise();
+    return swaggerResponse;
+  }
+
+  async retrieveSwaggerSummary(swaggerResponse: SwaggerUploadResponse): Promise <SwaggerSummary> {
+    console.log('Entered Swagger Summary Retriever');
+    let receivedSummary = false;
+    while (!receivedSummary) {
+      /* eslint-disable no-await-in-loop */
```

Comment:
This rule exists due to concerns about performance. So this should likely be refactored to not use await inside a while loop. Instead, what we should do would be to obtain all of the promises and then group them together with something like `Promise.all()` and await on that result outside of the loop.

Reply:
If we are not actually obtaining multiple responses, then we likely shouldn't be using a while loop.

My guess as to the reasoning would be to continuously retry until the summary is available, but I believe that continuously retrying forever is not recommended.

Reply:
After consulting with the backend team, we need this to continue to call the backend until we receive the one object we're awaiting.